src/asynk/async_scheduler.rs

```diff
@@ -0,0 +1,207 @@
+use crate::asynk::async_queue::AsyncQueueable;
+use crate::asynk::async_queue::PeriodicTask;
+use crate::asynk::Error;
+use async_recursion::async_recursion;
+use log::error;
+use std::time::Duration;
+use tokio::time::sleep;
+use typed_builder::TypedBuilder;
+
+#[derive(TypedBuilder)]
+pub struct Scheduler<'a> {
+    #[builder(setter(into))]
+    pub check_period: u64,
+    #[builder(setter(into))]
+    pub error_margin_seconds: u64,
+    #[builder(setter(into))]
+    pub queue: &'a mut dyn AsyncQueueable,
+    #[builder(default = 0, setter(into))]
+    pub number_of_restarts: u32,
+}
+
+impl<'a> Scheduler<'a> {
+    #[async_recursion(?Send)]
+    pub async fn start(&mut self) -> Result<(), Error> {
+        let task_res = self.schedule_loop().await;
```

Comment:
let's add 1 second of sleep after this step finishes.

so a system (maybe db) has time to recover

Reply:
Solve in [e1d31d2](https://github.com/ayrat555/fang/pull/47/commits/e1d31d215656077b385a07f4582a30a6d595a3c5)